### ISSUE TITLE
Unreachable "else block" code isn't reported as such, when "is var" is used

### ISSUE THREAD

**Issue by DavidArno** [dotnet/roslyn#15489]
**Version Used**: VS2017 RC

**Steps to Reproduce**:
````cs
class Program
{
    static void Main(string[] args)
    {
        var s = "test";
        if (s is var x)
        {
            Console.WriteLine($"x = {x}");
        }
        else
        {
            Console.WriteLine("s isn't a var!"); // unreachable code
        }
    }
}
````

**Expected Behavior**:
As `s is var x` will always be true, an "Unreachable code detected" warning should be reported for the contents of the `else` block.

**Actual Behavior**:
No warning is reported.

**Comment by gafter** [comment]
As currently specified, an unreachable code warning is only produced when the if expression is the constant `true`, not when it is a non-constant expression that is always true. However, this is a reasonable feature request.

**Comment by gafter** [comment]
@MadsTorgersen Does this belong on the LDM agenda?

**Comment by alrz** [comment]
I think there was a proposal for disallowing `e is var` because that's pointless! IMO declaration expressions are a better alternative when you just want to introduce a variable in an expression context.

**Comment by gafter** [comment]
The LDM met yesterday and confirmed that the current behavior is intended. We do not make any reachability accommodation for an is-pattern-matching operation that is irrefutable.
